core/src/err.rs

```diff
@@ -6,6 +6,23 @@ use constants::{
 use mongodb::error::{ErrorKind, WriteFailure};
 use thiserror::Error;
 
+#[derive(Debug, Clone)]
+pub struct Diagnostics {
```

Comment:
[nit] Would you be Ok renaming it QueryDiagnostics for newbies to avoid confusion with the ODBC diagnostics?
```suggestion
pub struct QueryDiagnostics {
```